event_model/__init__.py

```diff
@@ -119,6 +119,74 @@ def bulk_datum(self, doc):
         self.datum_page(bulk_datum_to_datum_page(doc))
 
 
+class Filler:
+    """Pass documents through, loading any externally-referenced data."""
+    ATTEMPTS = 10
+
+    def __init__(self, handler_registry, *,
+                 handler_cache=None, datum_cache=None):
+        self.handler_registry = handler_registry
+        if handler_cache is None:
+            handler_cache = {}
+        if datum_cache is None:
+            datum_cache = {}
+        self.handlers = handler_cache
+        self.datums = datum_cache
+
+    def __call__(self, name, doc):
+        return name, getattr(self, name)(doc)
```

Comment:
This either needs to have a stop method or more magic for if a doc is not returned

Reply:
Ha!  I had the same thought, wrote the comment, and then deleted it before posting :-p